nvflare/fuel/hci/server/file_transfer.py

```diff
@@ -315,3 +323,18 @@ def download_job(self, conn: Connection, args: List[str]):
     def info(self, conn: Connection, args: List[str]):
         conn.append_string("Server Upload Destination: {}".format(self.upload_dir))
         conn.append_string("Server Download Source: {}".format(self.download_dir))
+
+    def _authorize_submission(self, conn: Connection, args: List[str]):
+
+        folder_name = args[1]
+        zip_b64str = args[2]
+        data_bytes = convert_legacy_zip(b64str_to_bytes(zip_b64str))
+        conn.set_prop(ConnProps.JOB_DATA, data_bytes)
+
+        meta_file = f"{folder_name}/{META_FILE}"
+        with ZipFile(BytesIO(data_bytes), "r") as zf:
```

Comment:
for apps submitted as jobs without meta.json, does this still work?

Reply:
this still work, because this one 
data_bytes = convert_legacy_zip(b64str_to_bytes(zip_b64str))
is will add the default meta.json if one is missing